### PR TITLE
PIMS-1754 Property Delete

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: frontend/src/assets/images/close-square.svg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<svg version="1.2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" overflow="visible" preserveAspectRatio="none" viewBox="0 0 40 40" xml:space="preserve" y="0px" x="0px" id="Layer_1_1588366553327" width="38" height="38"><g transform="translate(1, 1)"><style type="text/css">
```
Comment (devinleighsmith) [inline frontend/src/assets/images/close-square.svg:1]: Christine custom made this icon, can't use react-icons.

File: frontend/src/components/common/GenericModal.tsx [hunk 2]
```diff
@@ -57,7 +57,7 @@ const GenericModal = (props: ModalProps) => {
 
   return (
     <Container>
-      <Modal show={show} onHide={ok}>
+      <Modal show={show} onHide={close}>
```
Comment (devinleighsmith) [inline frontend/src/components/common/GenericModal.tsx:60]: the default when the modal hides is whatever is configured for cancel.

File: frontend/src/forms/subforms/PagedBuildingForms.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ const PagedBuildingForms: React.FC<PagedBuildingFormsProps> = (props: PagedBuild
                 {...pagedBuildings}
               />
             </span>
-            <h4>Building Information</h4>
+            {!!formikProps.values.buildings.length && <h4>Building Information</h4>}
```
Comment (devinleighsmith) [inline frontend/src/forms/subforms/PagedBuildingForms.tsx:91]: Header should only display if buildings are present.

File: frontend/src/pages/SubmitProperty.scss [hunk 4]
```diff
@@ -14,17 +13,42 @@
...
   }
   .sideMap {
-    display: relative;
     padding: 0;
   }
   .form {
     background-color: #f2f2f2;
     padding: 1rem 3rem;
     max-height: calc(100vh - 70px - 45px - 42px);
+    height: calc(100vh - 70px - 45px - 42px);
```
Comment (devinleighsmith) [inline frontend/src/pages/SubmitProperty.scss:22]: ensure that the empty page renders spinners properly.

File: frontend/src/pages/SubmitProperty.tsx [hunk 5]
```diff
@@ -109,36 +118,20 @@ const SubmitProperty = (props: any) => {
...
-                if (!formDisabled && isStorageInUse(PARCEL_STORAGE_NAME)) {
-                  setShowSaveDraftDialog(true);
-                } else {
-                  history.goBack();
-                }
-              }}
-            >
-              <FaTimes size={20} />
-            </Button>
+            <span className="propertyButtons">
```
Comment (devinleighsmith) [inline frontend/src/pages/SubmitProperty.tsx:121]: extracting subcomponents out for readability.

File: frontend/src/actionCreators/parcelsActionCreator.tsx [hunk 6]
```diff
@@ -105,3 +105,18 @@ export const updateParcel = (parcel: IParcel) => (dispatch: Function) => {
...
+  return CustomAxios()
+    .delete(ENVIRONMENT.apiUrl + API.PARCEL_ROOT + `/${parcel.id}`, { data: parcel })
+    .then((response: AxiosResponse) => {
+      dispatch(success(actionTypes.DELETE_PARCEL, response.status));
+      dispatch(hideLoading());
+    })
+    .catch((axiosError: AxiosError) => {
+      dispatch(error(actionTypes.DELETE_PARCEL, axiosError?.response?.status, axiosError));
+    })
+    .finally(() => dispatch(hideLoading()));
```
Comment (goemen) [inline frontend/src/actionCreators/parcelsActionCreator.tsx:121]: Not that it should be done now, but I think we should plan on moving the loading progress dispatch to the axios interceptor.

**Comment by asanchezr** [comment]
@devinleighsmith you'll have to rebase and update the test snapshots before merging this.

**Pull request merged**